code/modules/vending/_vending.dm

```diff
@@ -1396,26 +1425,26 @@
  * product_to_vend - the product record of the item we're trying to vend
  * price_to_use - price of the item we're trying to vend
  */
-/obj/machinery/vending/proc/proceed_payment(obj/item/card/id/paying_id_card, datum/data/vending_product/product_to_vend, price_to_use)
+/obj/machinery/vending/proc/proceed_payment(obj/item/card/id/paying_id_card, datum/data/vending_product/product_to_vend, price_to_use, mob/living/alive_user)
 	var/datum/bank_account/account = paying_id_card.registered_account
 	if(account.account_job && account.account_job.paycheck_department == payment_department)
 		price_to_use = max(round(price_to_use * DEPARTMENT_DISCOUNT), 1) //No longer free, but signifigantly cheaper.
 	if(coin_records.Find(product_to_vend) || hidden_records.Find(product_to_vend))
 		price_to_use = product_to_vend.custom_premium_price ? product_to_vend.custom_premium_price : extra_price
 	if(LAZYLEN(product_to_vend.returned_products))
 		price_to_use = 0 //returned items are free
-	if(price_to_use && !account.adjust_money(-price_to_use, "Vending: [product_to_vend.name]"))
+	if(price_to_use && attempt_charge(src, alive_user, price_to_use) & COMPONENT_OBJ_CANCEL_CHARGE)
```

Comment:
```suggestion
	if(price_to_use && (attempt_charge(src, alive_user, price_to_use) & COMPONENT_OBJ_CANCEL_CHARGE))
```
Be careful with bitflag operations, they might own you in byond if you're not careful
Also, is this proc safe if passed a `null` user?